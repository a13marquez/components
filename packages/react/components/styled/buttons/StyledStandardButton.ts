import styled from 'styled-components';

export const StyledStandardButton = styled.button`
  font-size: 20px;
  line-height: 25px;
`