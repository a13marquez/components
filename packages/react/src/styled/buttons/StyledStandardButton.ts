import styled from 'styled-components';

const StyledStandardButton = styled.button`
  font-size: 20px;
  line-height: 25px;
`;

export default StyledStandardButton;