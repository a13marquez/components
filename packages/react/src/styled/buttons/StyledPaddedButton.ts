import styled from 'styled-components';
import StyledStandardButton from './StyledStandardButton';

const StyledPaddedButton = styled(StyledStandardButton)`
  margin-top: 24px;
  padding-top: 16px;
  padding-bottom: 16px;
  width: 100%;
  border-radius: 4px;
  background: #000;
  color: #ffffff;
  font-size: 16px;
  font-weight: bold;
  line-height: 14px;
  cursor: pointer;
`;

export default StyledPaddedButton;