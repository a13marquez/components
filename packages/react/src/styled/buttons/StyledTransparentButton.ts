import styled from 'styled-components';
import StyledStandardButton from './StyledStandardButton';

const StyledTransparentButton = styled(StyledStandardButton)`
  padding: 0 8px;
  height: 40px;
  border: none;
  background: transparent;
  color: ${props => props.color || '#000'};
  font-weight: bold;
  cursor: pointer;
`;

export default StyledTransparentButton;
