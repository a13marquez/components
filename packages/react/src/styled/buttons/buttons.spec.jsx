import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import StyledStandardButton from './StyledStandardButton';
import StyledTransparentButton from './StyledTransparentButton';
import StyledPaddedButton from './StyledButton';


test('It renders Styled Standard Button ', () => {
  const StandardButton = mount(<StyledStandardButton />);
  expect(StandardButton).toMatchSnapshot();
});

test('It renders Styled Count Button ', () => {
  const CountButton = mount(<StyledTransparentButton />);
  expect(CountButton).toMatchSnapshot();
});

test('It renders Styled Checkout Button ', () => {
  const CheckoutButton = mount(<StyledPaddedButton />);
  expect(CheckoutButton).toMatchSnapshot();
});

test('Styled Standard button have the correct styles', () => {
  const StandardButton = mount(<StyledStandardButton />);
  expect(StandardButton).toHaveStyleRule('font-size', '20px');
  expect(StandardButton).toHaveStyleRule('line-height', '25px');
});

test('Styled Count button have the correct styles', () => {
  const CountButton = mount(<StyledTransparentButton />);
  expect(CountButton).toHaveStyleRule('font-size', '20px');
  expect(CountButton).toHaveStyleRule('line-height', '25px');
  expect(CountButton).toHaveStyleRule('padding', '0 8px');
  expect(CountButton).toHaveStyleRule('height', '40px');
  expect(CountButton).toHaveStyleRule('border', 'none');
  expect(CountButton).toHaveStyleRule('background', 'transparent');
  expect(CountButton).toHaveStyleRule('color', '#000');
  expect(CountButton).toHaveStyleRule('font-weight', 'bold');
  expect(CountButton).toHaveStyleRule('cursor', 'pointer');
});

test('Styled Checkout button have the correct styles', () => {
  const CheckoutButton = mount(<StyledPaddedButton />);
  expect(CheckoutButton).toHaveStyleRule('margin-top', '24px');
  expect(CheckoutButton).toHaveStyleRule('padding-top', '16px');
  expect(CheckoutButton).toHaveStyleRule('padding-bottom', '16px');
  expect(CheckoutButton).toHaveStyleRule('width', '100%');
  expect(CheckoutButton).toHaveStyleRule('border-radius', '4px');
  expect(CheckoutButton).toHaveStyleRule('background', '#000');
  expect(CheckoutButton).toHaveStyleRule('color', '#fff');
  expect(CheckoutButton).toHaveStyleRule('font-size', '16px');
  expect(CheckoutButton).toHaveStyleRule('font-weight', 'bold');
  expect(CheckoutButton).toHaveStyleRule('line-height', '14px');
  expect(CheckoutButton).toHaveStyleRule('cursor', 'pointer');
});
