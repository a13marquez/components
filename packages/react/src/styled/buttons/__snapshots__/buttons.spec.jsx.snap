// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders Styled Checkout Button  1`] = `ReactWrapper {}`;

exports[`It renders Styled Count Button  1`] = `ReactWrapper {}`;

exports[`It renders Styled Standard Button  1`] = `ReactWrapper {}`;
